src/briefcase/platforms/android/apk.py

```diff
@@ -1 +1,205 @@
+            / "debug"
+            / "app-debug.apk"
+        )
+
+    def distribution_path(self, app):
+        return self.binary_path(app)
+
+    def verify_tools(self):
+        """
+        Verify that we're on a Python version where a support package URL is available.
+        """
+        if self.python_version_tag != "3.7":
+            raise BriefcaseCommandError(
+                """
+Found Python version {self.python_version_tag}. Android packaging currently requires Python 3.7.
+        """.format(
+                    self=self
+                )
+            )
+
+
+# The way to split this up is that you might start with more
+# stuff externally specified.
+
+
+class ApkCreateCommand(ApkMixin, CreateCommand):
+    description = "Create and populate an Android APK."
+
+    # TODO: Move these two into my app, and out of this pull request!
+    @property
+    def support_package_url(self):
+        "The URL of the support package to use for apps of this type."
+        # Someday this may move into the briefcase-support.org web app, probably
+        # around the time we support multiple Python versions.
+        return "https://github.com/paulproteus/Python-Android-support/releases/download/v0.5/3.7.zip"
+
+    @property
+    def app_template_url(self):
+        "The URL for a cookiecutter repository to use when creating apps"
+        # Add briefcase.toml to the cookiecutter template
+        return "https://github.com/paulproteus/cookiecutter-beeware-android.git"
+
+
+class ApkUpdateCommand(ApkMixin, UpdateCommand):
+    description = "Update an existing Android APK."
+
+
+class ApkBuildCommand(ApkMixin, BuildCommand):
+    description = "Build an Android APK."
+
+    # e.g.
+    # mock self.download_url to always download a sentinel zip
+    # - validate we create the android_sdk path
+    # see xcode tests for mocks of command line tools
+
+    @property
+    def android_sdk_url(self):
+        """The Android SDK URL appropriate to this operating system."""
+        # These URLs have existed since approximately 2017, and they have a built-in autoupdater,
+        # so I expect them to keep working for quite a few years longer.
+        return "https://dl.google.com/android/repository/sdk-tools-{os}-4333796.zip".format(
+            os=self.host_os.lower()
+        )
+
+    def verify_tools(self):
+        super().verify_tools()
+
+        print()
+        print("Ensuring we have the Android SDK...")
+        if not self.android_sdk_path.exists():
+            try:
+                android_sdk_zip_path = self.download_url(
+                    url=self.android_sdk_url,
+                    download_path=Path.home() / ".briefcase" / "tools",
+                )
+            except requests_exceptions.ConnectionError:
+                raise NetworkFailure("downloading Android SDK")
+            with ZipFile(android_sdk_zip_path) as android_sdk_zip:
+                android_sdk_zip.extractall(path=self.android_sdk_path_tmp)
```

Comment:
Any reason you're using an explicitly specified, constant tmp path, rather than a genuine tmpfile generated, unique-every-time directory? If we're going to go to the trouble of caring about the edge case of a "partially unrolled zipfile" being a problem, then using the same temp dir every time would seem to be a risk.

Reply:
Just a quick initial implementation. For the next revision, I am planning on doing `rmtree` on the `_tmp` filename, rather than creating temporary paths; that way, I avoid creating files in the filesystem that we never clean up. I can leave a comment explaining that rationale.

I'm also okay using a unique filename, but I think that it's a substantial virtue if we're able to clean up after ourselves, in which case there's no real upside to the random name IMHO.

Reply:
I suppose the bigger question is why the tmp version is needed at all. If the zip fails to unpack, then I don't see that having a temporary directory actually saves us from anything - we're going to need to do a cleanup. If it unpacks successfully, an in-situ unpack works; if ti doesn't, we should be purging the "bad" directory as part of the "it failed" cleanup.

Reply:
I guess I'm scared of the case where we can't run our own cleanup for some reason. I'm OK ignoring that case for now, and focusing on writing good cleanup. :)

Reply:
I tend to love the pattern where "if it exists, it works", that I've learned from some well-written shell scripts and Debian tools; that's where this comes from for me.